src/omnisharp/launcher.ts

```diff
@@ -161,21 +161,15 @@ export interface LaunchResult {
 
 export function launchOmniSharp(details: LaunchDetails): Promise<LaunchResult> {
     return new Promise<LaunchResult>((resolve, reject) => {
-        try {
-            return launch(details).then(result => {
-                // async error - when target not not ENEOT
-                result.process.on('error', reject);
-
-                // success after a short freeing event loop
-                setTimeout(function () {
-                    resolve(result);
-                }, 0);
-            }, err => {
-                reject(err);
-            });
-        } catch (err) {
-            reject(err);
-        }
+        launch(details).then(result => {
+            // async error - when target not not ENEOT
+            result.process.on('error', reject);
+
+            // success after a short freeing event loop
+            setTimeout(function () {
+                resolve(result);
+            }, 0);
+        });
```

Comment:
It's intended that rejection will be propagated to the caller, which is in server.ts [here](https://github.com/OmniSharp/omnisharp-vscode/pull/807/files#diff-f08c23c62ed01212f1e17a9530885a70R302).